tree/treeplayer/src/TBranchProxy.cxx

```diff
@@ -275,13 +318,22 @@ bool ROOT::Detail::TBranchProxy::Setup()
       // its mother's name
       fBranch = fDirector->GetTree()->GetBranch(fBranchName.Data());
       if (!fBranch) {
-         // FIXME
-         // While fixing ROOT-10019, this error was added to give to the user an even better experience
-         // in presence of a problem.
-         // It is not easy to distinguish the cases where this error is "expected"
-         // For now we do not print anything - see conversation here: https://github.com/root-project/root/pull/3746
-         //auto treeName = fDirector->GetTree()->GetName();
-         //::Error("TBranchProxy::Setup", "%s", Form("Unable to find branch %s in tree %s.\n",fBranchName.Data(), treeName));
+         if (!AreThereSubBranches(fBranchName.View(), *fDirector->GetTree())) {
+            // The next error refers specifically to the situation where the branch identified by fBranchName
+            // is not present and that is not expected. An example is when traversing a chain of files, the branch
+            // is missing from the file that we are switching into.
+            // Conversely, there are situations where the missing branch is indeed expected. A notable example is in
+            // the skeleton for analysis created by `TTree::MakeProxy`. If the TTree contains a split object, the branch
```

Comment:
The case is only related to `MakeProxy` in that the file used for one of the `MakeProxy` in roottest contains the pattern we are talking about.  So I would either remove this sentence or be more specific about which test this ROOT file is part of.

Reply:
I prefer removing the mention of MakeProxy and just describe the TTree structure that can lead to that situation